required/latex-lab/latex-lab-mathcapture.dtx

```diff
@@ -0,0 +1,505 @@
+%   \meta{tokens}, which will receive the environment type as |#1| and
+%   the content as |#2|.
+% \end{function}
+%
+% \subsection{Document level interfaces}
+%
+% \begin{function}{\RegisterMathEnvironment}
+%   \begin{syntax}
+%     \cs{RegisterMathEnvironment} \oarg{options} \Arg{env}
+%   \end{syntax}
+%   Registers the \meta{env} as a math environment which should be captured
+%   and made available. This is necessary for all top-level math mode
+%   environments: low-level errors may result if these are not correct
+%   set up. One or more key--value \meta{options} may also be given:
+%   \begin{itemize}
+%     \item[\texttt{arg-spec}] The argument specification taken by the
+%       beginning of the environment; this is used to remove non-mathematical
+%       material.
+%   \end{itemize}
+% \end{function}
+%
+% \MaybeStop{\setlength\IndexMin{200pt}  \PrintIndex  }
+%
+% \section{The Implementation}
+%
+%    \begin{macrocode}
+%<@@=math>
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+%<*kernel>
+%    \end{macrocode}
+%
+% \subsection{File declaration}
+%    \begin{macrocode}
+\ProvidesFile{latex-lab-mathcapture.ltx}
+        [2023-01-06 v0.1a Grab all the math(s)]
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\ExplSyntaxOn
+%    \end{macrocode}
+%
+% \subsection{Setup}
+%
+% Loading \pkg{amsmath} is an absolute requirement: this avoids needing to
+% have conditional definitions and deals with how to define \cs{[}/\cs{]}
+% neatly.
+%    \begin{macrocode}
+\tl_gput_right:Nn \@kernel@before@begindocument
+  { \RequirePackage { amsmath } }
+%    \end{macrocode} 
+%
+% \subsection{Data structures}
+%
+% \begin{variable}{\l_@@_collected_bool}
+%   Tracks if math mode material has been collected, which happens inside
+%   \pkg{amsmath} environments as well as those handled directly here.
+%    \begin{macrocode}
+\bool_new:N \l_@@_collected_bool
+%    \end{macrocode}
+% \end{variable}
+%
+% \subsection{Interface commands}
+%
+% \begin{macro}
+%   {\@@_process:nn, \@@_process:Vn, \@@_process_auxi:nn, \@@_process_auxii:nn}
+%   A no-op place-holder; the internal wrapper means that it does not need to
+%   be concerned with internals.
+%    \begin{macrocode}
+\cs_new_protected:Npn \@@_process:nn #1#2
+  {
+    \legacy_if:nF { measuring@ }
+      { \tl_trim_spaces_apply:nN {#2} \@@_process_auxi:nn {#1} }
+  }
+\cs_generate_variant:Nn \@@_process:nn { V }
+\cs_new_protected:Npn \@@_process_auxi:nn #1#2
+  { \@@_process_auxii:nn {#2} {#1} }
+\cs_new_protected:Npn \@@_process_auxii:nn #1#2 { }
+%    \end{macrocode}
+% \end{macro}
+%
+% \begin{macro}{\math_processor:n}
+%   A simple installer
+%    \begin{macrocode}
+\cs_new_protected:Npn \math_processor:n #1
+  { \cs_set_protected:Npn \@@_process_auxii:nn ##1##2 {#1} }
+%    \end{macrocode}
+% \end{macro}
+%
+% \subsection{Content grabbing}
+%
+% \begin{macro}{\@@_grab_dollar:w}
+%   Grab up to a single |$|, for inline math mode, suppressing
+%   any processing if the first token is \tn{m@th}.
```

Comment:
Typo, should be:  
   . . . of the first . . .

Reply:
Pretty sure I mean `if` here :)

Reply:
So you do!   Must have been somewhere else that needed this:-).